stdlib/seq.mli

```diff
@@ -13,97 +13,807 @@
+     consume the shorter sequence, and ignore the remainder of the longer
+     sequence.
+   - [exists2], [equal], [compare] consume the sequences down to a certain
+     depth, which is a priori unpredictable.
+
+   None of the functions in this section is lazy. These functions
+   are consumers: they force some computation to take place. *)
+
+val is_empty:  'a t -> bool
+(** [is_empty xs] determines whether the sequence [xs] is empty.
+
+    @since 4.14 *)
+
+val head : 'a t -> 'a
+(** [head xs] is the first element of the sequence [xs].
+
+    The sequence [xs] must be nonempty.
+
+    @raise Invalid_argument if [xs] is empty.
+    @since 4.14 *)
+
+val tail : 'a t -> 'a t
+(** [tail xs] is the sequence [xs], deprived of its first element.
+
+    The sequence [xs] must be nonempty.
+
+    @raise Invalid_argument if [xs] is empty.
+    @since 4.14 *)
+
+val uncons : 'a t -> ('a * 'a t) option
+(** If [xs] is empty, then [uncons xs] is [None].
+
+    If [xs] is nonempty, then [uncons xs] is
+    [Some (head xs, tail xs)],
+    that is, a pair of the head and tail of the sequence [xs].
+
+    This equivalence holds if [xs] is persistent.
+    If [xs] is ephemeral, then [uncons] must be preferred
+    over separate calls to [head] and [tail],
+    which would cause [xs] to be queried twice.
+
+    @since 4.14 *)
+
+
+
+val length : 'a t -> int
+(** [length xs] is the length of the sequence [xs].
+
+    The sequence [xs] must be finite.
+
+    @since 4.14 *)
+
+val iter : ('a -> unit) -> 'a t -> unit
+(** [iter f xs] invokes [f x] successively
+    for every element [x] of the sequence [xs],
+    from left to right.
+
+    The sequence [xs] must be finite. *)
+
+val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b t -> 'a
+(** [fold_left f _ xs] invokes [f _ x] successively
+    for every element [x] of the sequence [xs],
+    from left to right.
+
+    An accumulator of type ['a] is threaded through the calls to [f].
+
+    The sequence [xs] must be finite. *)
+
+val iteri : (int -> 'a -> unit) -> 'a t -> unit
+(** [iteri f xs] invokes [f i x] successively
+    for every element [x] located at index [i] in the sequence [xs].
+
+    The sequence [xs] must be finite.
+
+    [iteri f xs] is equivalent to
+    [iter (fun (i, x) -> f i x) (zip (ints 0) xs)].
+
+    @since 4.14 *)
+
+val fold_lefti : (int -> 'b -> 'a -> 'b) -> 'b -> 'a t -> 'b
+(** [fold_lefti f _ xs] invokes [f _ i x] successively
+    for every element [x] located at index [i] of the sequence [xs].
+
+    An accumulator of type ['a] is threaded through the calls to [f].
+
+    The sequence [xs] must be finite.
+
+    [fold_lefti f accu xs] is equivalent to
+    [fold_left (fun accu (i, x) -> f accu i x) accu (zip (ints 0) xs)].
+
+    @since 4.14 *)
+
+val for_all : ('a -> bool) -> 'a t -> bool
+(** [for_all p xs] determines whether all elements [x] of the sequence [xs]
+    satisfy [p x].
+
+    The sequence [xs] must be finite.
```

Comment:
In this specific case "the traversal may or may not terminate on infinite sequences, depending on the function's input"